File: 크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    stack = []
    for i in moves :
        for j in range (len(board)) :
            if board[j][i-1] != 0 : #0이 아닐때
                if len(stack) > 0 and stack[-1] == board[j][i-1] : #서로 같은숫자일 시
                    stack.pop()
                    answer+=2
                    board[j][i-1] = 0
                    break
                else :
                    stack.append(board[j][i-1])
                    board[j][i-1] = 0
                    break
    return answer

File: lv1/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < moves.length; i++) {
            for (int j = 0; j < board.length; j++) {
                if (board[j][moves[i]-1] != 0) {
                    stack.add(board[j][moves[i]-1]);
                    board[j][moves[i]-1] = 0;
                    if (stack.size() > 1) {
                        if (stack.get(stack.size() - 2) == stack.lastElement()) {
                            answer+=2;
                            stack.pop();
                            stack.pop();
                        }
                    }
                    break;
                }
            }
        }
        return answer;
    }
}
